### ISSUE TITLE
please make boot process "synchronous" + provide simple logging with "log_file", "log_level" options.

### ISSUE THREAD

**Issue by faaliyet** [chrislusf/seaweedfs#153]
the default boot process is asynchronous and does not allow developers to point **ports to volumes** like this;

* 7001 -> volume_1
* 7002 -> volume_2
* 7003 -> volume_3
* 7004 -> volume_4
so on..

I handle file routes in backends , so I need to do this **programmatically**. I've also tried with "& wait & " in shell but the result is same.

As I point also here; https://github.com/chrislusf/seaweedfs/issues/107
program waits for volumes to be ready, and logs info. until they are ready. You said, it's ok, but it's not ok for **real world apps**.

devops are not always logging "infos", for high traffic websites, devops prefer to *disable logs for infos* , and activate them for more critical things like warnings, fatal errors, so on...
And they *trigger* that logs with *sms, email*... 

Another thing that I want to discuss is; logging operation itself. I've looked into the default log option, and found that "not productive". Why not we can't use simple logging like another daemons ? 
**log_level** *(warning, info, error)* , **log_file** *(path_to_file)* and that's all. So, we can integrate it into centralized logs like *logstash* ...

```
In short, please;
1. Make boot process "synchronous".
2. Provide simple logging with "log_file", "log_level" options.

Is it only me who think about the above improvements or we are more ?
+1 if you agree.
```

**Thanks** for this open source super file system.

**Comment by faaliyet** [comment]
```
volume_size_in_mb="$((${volume_size} * 1000))"

command="./weed master -mdir=\"${directory_storage}/seaweedfs/meta/\" -port=${port_master} -volumeSizeLimitMB=${volume_size_in_mb}"

i="0"
while [ "${i}" -lt "${volume_number}" ]; do
	i=$[$i+1]

	port_volume=$[$port_volume+1]

	command+=" & wait & ./weed volume -dir=\"${directory_storage}/seaweedfs/vol_${i}/\" -max=1 -mserver=\"localhost:${port_master}\" -port=${port_volume} -images.fix.orientation=false -leveldb=${leveldb}"
done

eval $(echo $command)
```

**Comment by chrislusf** [comment]
2. Seaweedfs uses glog. Try "weed -h" to see the options. Does it help?

**Comment by faaliyet** [comment]
Sorry, I didn't understand. can you clarify a little big ? yes, I've checked "weed -h" help options, but didn't/can't find anything useful for my use case. And may it's already there the solution of my problem, but I just can't see that. You might help for this. :+1:

**Comment by faaliyet** [comment]
If you talk about the logging option, I know it exists, what I mean is, making it simple as my explanation above. keeping logging boot process is not something useful. anyway I'm just sharing my ideas.

**Comment by chrislusf** [comment]
You can use -v=4 to change log levels.

**Comment by chrislusf** [comment]
Regarding "making boot synchronous":
1. When starting master server, if you can clean the master meta folder specified by "-mdir", it will be much faster to startup.
2. It's better to have multiple volumes for one volume server. So you do not need to have such a complicated startup script.

**Comment by faaliyet** [comment]
```2. It's better to have multiple volumes for one volume server. So you do not need to have such a complicated startup script.```

Well, thanks for your response. It's not that complicated. However, it's how I create port-volume relationship. Do you have any suggestion how to do it better? Let's say I have 20 volumes and then
port 7000 - as a namespace for volume 1-5,  7100 - as a namespace for volume 6-10, and so on...
```
7000 - volume_server_1 - 1,2,3,4,5
7100 - volume_server_2 - 6,7,8,9,10
```
The problem is, in backends, I'm not asking SeaweedFS for a file location, I'm directly sending a file to the specific volume server. And I didn't find any solution yet. May, you can put another flag/argument where we start volumes or something else option to allow user to create port-volume-volume-server relationship. Any ideas? :)

**Comment by faaliyet** [comment]
I think, volume id;s should not be randomized at startup, that's the core of the problem. So, if they keep starting asynchronously, then they will be randomized always. correct me, if I'm wrong.

**Comment by faaliyet** [comment]
How about something like this?

./weed volume -dir="/data/seaweedfs/vol_1/" -max=7 -mserver="localhost:7000" -port=7001 -images.fix.orientation=false -leveldb=true" volume_ids="1,2,3,4,5,6,7"

so, we can be sure that; volume_ 7001 have 1,2,3,4,5,6,7. I think, I should not be hard to do this. You can also add some validations, for the new arg -volume_ids with -max arg.

**Comment by chrislusf** [comment]
Use this HTTP API to create a specific volume:

https://github.com/chrislusf/seaweedfs/wiki/API#create-%D0%B0-specific-volume-on-a-specific-volume-server

**Comment by faaliyet** [comment]
It seems OK to me, but how then I can restart master-slaves?

**Comment by faaliyet** [comment]
the same way as I start volumes? some options are confused :S

**Comment by faaliyet** [comment]
https://github.com/chrislusf/seaweedfs/wiki/Failover-Master-Server#startup-multiple-master-servers ?

**Comment by chrislusf** [comment]
I removed the second section "when-you-need-to-restart". It is confusing. Just follow this section:

https://github.com/chrislusf/seaweedfs/wiki/Failover-Master-Server#cheat-sheet-startup-multiple-servers

**Comment by faaliyet** [comment]
Thanks ! I'll re-code something new today to see that my expectation solves the problem. So, I can back here to close this issue asap :) Thanks.

**Comment by faaliyet** [comment]
right now, I'm back on the issue. Can you explain how can I use the followings together:
https://github.com/chrislusf/seaweedfs/wiki/Failover-Master-Server#cheat-sheet-startup-multiple-servers

https://github.com/chrislusf/seaweedfs/wiki/API#create-%D0%B0-specific-volume-on-a-specific-volume-server

help says: /weed volume starts volume, and that's fine. If we dont use **-max=** it automatically starts up with 7 volumes. Let's say we have 1 master 1 volume,  so we have 1,2,3,4,5,6,7 parts in volume server, so

how this helps in this scenario?
 https://github.com/chrislusf/seaweedfs/wiki/API#create-%D0%B0-specific-volume-on-a-specific-volume-server

as we already have 1,2,3,4,5,6,7..

Also, I've looked at glog doc and found that it does not allow to use same file for logging, so it's possible to have thousands of log files..

**Comment by faaliyet** [comment]
Your daemon is very good! **I respect your effort**, but I feel like I'm wasting my time. SeaweedFs is behind a reverse proxy, so I need something that is not randomized for each start-up.  port -> volume ->parts .. if I can't this map, it's just useless *for me*.

**Comment by faaliyet** [comment]
If you would like help me at this point, please provide me pseudo code, not doc please. I'm reading them over and over again, and I can't solve my problem with the exist docs. I don't know, may the solution is already there but I can't catch that. :S Any help is welcome..

**Comment by faaliyet** [comment]
Yeah, finally, I've skipped the docs, and just applied what you said. It works! 
curl "http://localhost:8080/admin/assign_volume?replication=000&volume=1" however, why returns {"error":""} ? I think, {"success":"Your command has been run successfully"} is more preferable. Anyway, **Thanks**, your help saved my day :+1:
